gui/backend/upsert_bot_config_test.go

```diff
@@ -0,0 +1,70 @@
+package backend
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stellar/kelp/gui/model2"
+	"github.com/stellar/kelp/support/kelpos"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsBotNameValid(t *testing.T) {
+	testCases := []struct {
+		name      string
+		botName   string
+		strategy  string
+		wantValid bool
+	}{
+		{
+			name:      "success default",
+			botName:   "George the Friendly Octopus",
+			wantValid: true,
+		},
+		{
+			name:      "success numeric",
+			botName:   "George the 1337 Octopus",
+			wantValid: true,
+		},
+		{
+			name:      "success dash",
+			botName:   "George-the-Friendly-Octopus",
+			wantValid: true,
+		},
+		{
+			name:      "failure <>",
+			botName:   "George<>the Friendly Octopus",
+			wantValid: false,
+		},
+		{
+			name:      "failure \\/",
+			botName:   "George\\/the Friendly Octopus",
+			wantValid: false,
+		},
+	}
+
+	for _, k := range testCases {
+		t.Run(k.name, func(t *testing.T) {
+			// errors would be thrown by the `regex` package, not due to kelp logic
+			gotValid, gotErr := isBotNameValid(k.botName)
+			assert.NoError(t, gotErr)
+			assert.Equal(t, k.wantValid, gotValid)
+			// early return on unsupported bot names, so we do not write an invalid filename
+			if !k.wantValid {
+				return
+			}
+
+			// confirm that we can write and delete an empty file to the bot's filepath
+			filenamePair := model2.GetBotFilenames(k.botName, "buysell")
+			fileBase, e := kelpos.MakeOsPathBase()
```

Comment:
we don't want to use `MakeOsPathBase()` here. This will write to the user's production directory!
we should instead use a temporary directory like [`os.TempDir`](https://golang.org/pkg/os/#TempDir).